### PR TITLE
[WIP] Add Model class for Hugging Face Transformers T5

### PR DESCRIPTION
**Pull request by craffel**
Currently experimental, use at your own risk!

[Here](https://gist.github.com/craffel/cb62f4d5787cda1837dc7fe2e3f35d03) is an example showing fine-tuning T5-Base on CoLA. It gets to ~62% accuracy after 5000 steps. I think we got 63% with Mesh TensorFlow at some point, but that was saving checkpoints more often and also training for a lot longer. This example also uses a different optimizer (so we can't reuse pretraining optimizer state). Given all of that I would definitely call the Hugging Face pipeline ready-to-go for fine-tuning (though it will never be useful for exactly replicating our results).

TODO:
- [x] Make `predict` take list of strings or files
- [x] Add `predict` method
- [x] Remove args from base `Model` class functions and reorder
- [x] Save a checkpoint at the end of training
- [x] Add option for evaluating all checkpoints
- [x] Add example code
- [x] Add `transformers` as a library dep
- [x] Add private getters for model and step

### CODE REVIEW

**Comment by craffel** [comment]
Superseded by #156
